File: OpenLibraryBackend/src/main/java/com/open/library/POJO/Publisher.java
package com.open.library.POJO;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.DynamicInsert;
import org.hibernate.annotations.DynamicUpdate;

import java.util.Collection;

@Data
@NoArgsConstructor
@AllArgsConstructor
@DynamicInsert
@DynamicUpdate
@Entity
@Table(name = "publishers")

public class Publisher {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "publisher_id")
    private Long id;

    private String name;

    private String address;

    @Column(columnDefinition = "TEXT")
    private String story;

    @OneToMany(mappedBy = "publisher")
    private Collection<Book> books;

    private boolean is_activated;

    private boolean is_deleted;
}

File: OpenLibraryFrontend/src/app/shared/models/Publisher.ts
export class Publisher {
    id!: number;
    name!: string;
    address!: string;
    story!: string;
    is_activated!: boolean;
    is_deleted!: boolean;
    constructor(id: number, name: string, address: string, story: string, is_activated: boolean, is_deleted: boolean) {
        this.id = id;
        this.name = name;
        this.address = address;
        this.story = story;
        this.is_activated = is_activated;
        this.is_deleted = is_deleted
    }
}